recommender/static/html/resourcebox.html

```diff
@@ -1,4 +1,4 @@
-<%page args="title,votes,url,description,descriptionText,id,isProblematic,problematicReason,voteMode,endorseMode,endorsedReason"/> 
+<%page args="title,votes,url,description,descriptionText,id,isProblematic,problematicReason,voteMode,endorseMode,endorsedReason" expression_filter="h"/>
```

Comment:
@awaisdar001 @yro: This may be the only change that is strictly necessary.  You may want to consider breaking this out into its own PR, and leaving all the other changes for follow-up (if necessary).  The other changes may require more discussion as well.

Reply:
Ignore this comment.  I understand this would be more work at this point. :)

Reply:
:-) 
and only `endorsedReason` is rendered here on this page, not `flag` or `recommender_title`. They need to be escaped at backend so that we do not store the data. 
 

@robrap Can you review and verify the fix on the sandbox?